core/indexer/elasticsearchDatabase.go

```diff
@@ -213,32 +259,59 @@ func (esd *elasticSearchDatabase) buildTransactionBulks(
 	return bulks
 }
 
-func (esd *elasticSearchDatabase) serializeBulkTx(bulk []*Transaction) bytes.Buffer {
-	var buff bytes.Buffer
-	for _, tx := range bulk {
-		meta := []byte(fmt.Sprintf(`{ "index" : { "_id" : "%s", "_type" : "%s" } }%s`, tx.Hash, "_doc", "\n"))
-		serializedTx, err := json.Marshal(tx)
+// SaveMiniblocks will prepare and save information about miniblocks in elasticsearch server
+func (esd *elasticSearchDatabase) SaveMiniblocks(header data.HeaderHandler, body *block.Body) {
+	miniblocks := esd.getMiniblocks(header, body)
+	if miniblocks == nil {
+		log.Warn("indexer: could not index miniblocks")
+		return
+	}
+	if len(miniblocks) == 0 {
+		return
+	}
+
+	buff := serializeBulkMiniBlocks(header.GetShardID(), miniblocks)
+	err := esd.dbWriter.DoBulkRequest(&buff, miniblocksIndex)
+	if err != nil {
+		log.Warn("indexing bulk of miniblock", "error", err.Error())
+	}
+}
+
+func (esd *elasticSearchDatabase) getMiniblocks(header data.HeaderHandler, body *block.Body) []*Miniblock {
+	headerHash, err := core.CalculateHash(esd.marshalizer, esd.hasher, header)
+	if err != nil {
+		log.Warn("indexer: could not calculate header hash", "error", err.Error())
+		return nil
+	}
+
+	encodedHeaderHash := hex.EncodeToString(headerHash)
+
+	miniblocks := make([]*Miniblock, 0)
+	for _, miniblock := range body.MiniBlocks {
+		mbHash, err := core.CalculateHash(esd.marshalizer, esd.hasher, miniblock)
 		if err != nil {
-			log.Debug("indexer: marshal",
-				"error", "could not serialize transaction, will skip indexing",
-				"tx hash", tx.Hash)
 			continue
 		}
-		// append a newline foreach element
-		serializedTx = append(serializedTx, "\n"...)
 
-		buff.Grow(len(meta) + len(serializedTx))
-		_, err = buff.Write(meta)
-		if err != nil {
-			log.Warn("elastic search: serialize bulk tx, write meta", "error", err.Error())
+		encodedMbHash := hex.EncodeToString(mbHash)
+
+		mb := &Miniblock{
+			Hash:            encodedMbHash,
+			SenderShardID:   miniblock.SenderShardID,
+			ReceiverShardID: miniblock.ReceiverShardID,
+			Type:            miniblock.Type.String(),
 		}
-		_, err = buff.Write(serializedTx)
-		if err != nil {
-			log.Warn("elastic search: serialize bulk tx, write serialized tx", "error", err.Error())
+
+		if mb.SenderShardID == header.GetShardID() {
+			mb.SenderBlockHash = encodedHeaderHash
```

Comment:
Is this done to make some links between headers and miniblocks to show in the explorer?

Reply:
yes. We want to link miniblocks in explorer to see when a miniblocks was included in sender shard and  also in receiver shard.

Reply:
ok